enclave-runtime/src/lib.rs

```diff
@@ -293,27 +302,51 @@ pub unsafe extern "C" fn call_rpc_methods(
 	sgx_status_t::SGX_SUCCESS
 }
 
-fn side_chain_rpc_int<PB, O>(request: &str, _ocall_api: O) -> Result<String>
+fn sidechain_rpc_int<PB, O>(request: &str, ocall_api: O) -> Result<String>
 where
 	PB: BlockT<Hash = H256>,
 	NumberFor<PB>: BlockNumberOps,
-	O: EnclaveOnChainOCallApi + 'static,
+	O: EnclaveAttestationOCallApi + EnclaveOnChainOCallApi + 'static,
 {
-	// Skip sidechain import now until #423 is solved.
-	// let _ = EnclaveLock::read_all()?;
-	//
-	// let header = LightClientSeal::<PB>::unseal()
-	// 	.map(|v| v.latest_finalized_header(v.num_relays()).unwrap())?;
-	//
-	// let importer: BlockImporter<AuthorityPair, PB, _, O, _> = BlockImporter::default();
-	//
-	// let io = side_chain_io_handler(move |signed_blocks| {
-	// 	import_sidechain_blocks::<PB, _, _, _>(signed_blocks, &header, importer.clone(), &ocall_api)
-	// });
-
-	let io = side_chain_io_handler::<_, crate::error::Error>(move |signed_blocks| {
-		log::info!("[sidechain] Imported blocks: {:?}", signed_blocks);
-		Ok(())
+	let _sidechain_lock = EnclaveLock::write_all()?;
+
+	let validator_access = ValidatorAccessor::<PB>::default();
+
+	let (latest_parentchain_header, genesis_hash) = validator_access.execute_on_validator(|v| {
+		let latest_parentchain_header = v.latest_finalized_header(v.num_relays())?;
+		let genesis_hash = v.genesis_hash(v.num_relays())?;
+		Ok((latest_parentchain_header, genesis_hash))
+	})?;
+
+	// FIXME: Where should all these things be defined? Not in the rpc interface I guess?
```

Comment:
That's actually a question for this review - I wasn't sure, because as far as I understood this whole function here is only temporary?

Reply:
In my PR #530 I introduced a generic component container that can be initialized once. And in the `enclave_runtime` a separate file `global_components.rs` where I put all the specific types used in the enclave and the global components. It probably makes sense to have an `TopPoolOperationHandler` and a `BlockImporter` instance there.

Reply:
Agreed on that one. So I'll change it once #530 is merged.

Reply:
So this is probably on the table again, since #530 is merged

Reply:
Yes, this is done now, resolving the conversation 👍